src/librustdoc/visit_lib.rs

```diff
@@ -68,7 +68,8 @@ impl<'a, 'tcx, 'rcx> LibEmbargoVisitor<'a, 'tcx, 'rcx> {
         }
 
         for item in self.cx.tcx.item_children(def_id).iter() {
-            if !item.is_import || item.vis == Visibility::Public {
+            if self.cx.tcx.def_key(item.def.def_id()).parent.map_or(false, |d| d != def_id.index) ||
```

Comment:
Oh, you have the condition inverted - `!is_import` means it was originally defined *in that module*, whereas `d != def_id.index` means the parent isn't the module itself, which means it was originally defined *elsewhere*.